guides/use-microprofile-amqp-connector-with-ssl-openshift.adoc

```diff
@@ -0,0 +1,195 @@
+[source, bash]
+----
+oc create secret generic brokeramqp-ssl-secret --from-file=broker.ks --from-file=client.ts --from-literal=keyStorePassword='changeit' --from-literal=trustStorePassword='changeit'
+----
+
+Then create file `broker.yaml` with following content referencing above secret:
+[source, yaml]
+----
+apiVersion: broker.amq.io/v1beta1
+kind: ActiveMQArtemis
+metadata:
+  name: amq-broker
+  application: amq-broker-app
+spec:
+  acceptors:
+    - port: 61617
+      verifyHost: false
+      needClientAuth: true
+      expose: true
+      multicastPrefix: jms.topic.
+      name: all
+      connectionsAllowed: 10
+      sslEnabled: true
+      protocols: all
+      sslSecret: brokeramqp-ssl-secret
+      sslProvider: JDK
+      anycastPrefix: jms.queue.
+  console:
+    expose: true
+  deploymentPlan:
+    journalType: nio
+    messageMigration: false
+    persistenceEnabled: false
+    requireLogin: false
+    size: 1
+  upgrades:
+    enabled: false
+    minor: false
+----
+
+To simplify this example, there is set `verifyHost: false` to avoid need to set correct hostname in the "client" certificate for reactive messaging application.
+
+Run following command to deploy AMQ 7 broker with secured acceptor:
+[source, shell]
+----
+oc create -f broker.yaml
+----
+
+== Build Reactive Messaging application using Wildfly Glow
+
+We will use example reactive messaging application in this guide that consists of a single `ProducingBean` and `ConsumingBean`.
+Where `ProducingBean` periodically produces messages into the channel and `ConsumingBean` consumes them. It also contains needed scripts and yaml files to set up AMQ 7 broker on Openshift.
+
+Example application is available on GitHub:
+[source, bash]
+----
+git clone git@github.com:mnovak1/guides.git
+cd guides/microprofile-reactive-messaging-amqp-connector-example/
+----
+
+The important part of the application is `microprofile-config.properties` file configuring secured AMQP connector to remote AMQ 7 broker:
+[source, bash, options="nowrap"]
+----
+amqp-host=<url-to-amq-broker-route>
+amqp-port=443
+amqp-username=admin
+amqp-password=admin
+amqp-use-ssl=true
+mp.messaging.connector.smallrye-amqp.wildfly.elytron.ssl.context=amqp-ssl-context
+
+mp.messaging.outgoing.source.connector=smallrye-amqp
+
+mp.messaging.incoming.in.connector=smallrye-amqp
+mp.messaging.incoming.in.address=source
+----
+
+[NOTE]
+--
+Update `amqp-host` - url of remote AMQ 7 broker deployed on Openshift, for example "amq-broker-all-0-svc-rte-mnovak.apps.mnovak-oegn.psi.domain.com". You can figure out this by checking `oc get routes` for `amq-broker-all-0-svc-rte` route.
+--
+
+`amqp-port` - port where AMQ 7 broker is listening, in case of openshift there is used port `443` for secured communication
+
+`amqp-username` and `amqp-password` - username and password for authentication to AMQ 7 broker. `admin/admin` is default used in AMQ 7 broker
```

Comment:
we are keeping default credentials for a broker that is exposed outside the cluster. This does not look like a good practice, wdyt?

Reply:
I think it's ok as there is exposed only acceptor with 2-way SSL. Server will trust only to client cert which is in its truststore configured by user. So even though it's self-signed cert, it should be safe.